src/main/java/Duke.java

```diff
@@ -62,17 +71,104 @@ public static void printTaskDone(String[] commands, int index, Task[] taskList)
-            System.out.println("   [" + taskList[taskDoneIndex - 1].getStatusIcon() + "] " + taskList[taskDoneIndex - 1].getDescription());
+            System.out.println("   " + taskList[taskDoneIndex - 1].toString());
             System.out.println("____________________________________________________________");
         }
     }
 
+    public static boolean checkValidEvent(String inWord) {
+        String[] commands = inWord.split("\\s+", 2);
+        if (commands.length != 2 || !inWord.contains(" /at")) {
+            return false;
+        }
+
+        String[] details = commands[1].split("/at", 2);
+        String descriptionDetails = details[0].trim();
+        String descriptionAt = details[1].trim();
+
+        if (details.length != 2 || descriptionDetails.isEmpty() || descriptionAt.isEmpty()) {
+            return false;
+        }
+        return true;
+    }
+
+    public static void printEvent(String inWord, int index, Task[] taskList) {
+        String[] commands = inWord.split("\\s+", 2);
+        String[] details = commands[1].split("/at", 2);
+
+        String description = details[0].trim();
+        String at = details[1].trim();
+
+        Event newItem = new Event(description, at);
+        taskList[index] = newItem;
+        System.out.println("____________________________________________________________");
+        System.out.println(" Got it. I've added this task:");
+        System.out.println("   " + newItem);
+        System.out.println(" Now you have " + (index + 1) +" tasks in the list.");
+        System.out.println("____________________________________________________________");
+    }
+
+    public static boolean checkValidTodo(String inWord) {
+        String[] commands = inWord.split("\\s+", 2);
+        String details = commands[1];
+        if (commands.length != 2 || details.isEmpty()) {
+            return false;
+        }
+        return true;
+    }
+
+    public static void printTodo(String inWord, int index, Task[] taskList) {
+        String[] commands = inWord.split("\\s+", 2);
+        String description = commands[1];
+
+        Todo newItem = new Todo(description);
+        taskList[index] = newItem;
+        System.out.println("____________________________________________________________");
+        System.out.println(" Got it. I've added this task:");
+        System.out.println("   " + newItem);
+        System.out.println(" Now you have " + (index + 1) +" tasks in the list.");
+        System.out.println("____________________________________________________________");
+    }
+
+    public static boolean checkValidDeadline(String inWord) {
+        String[] commands = inWord.split("\\s+", 2);
+        if (commands.length != 2 || !inWord.contains(" /by")) {
+            return false;
+        }
+
+        String[] details = commands[1].split("/by", 2);
+        String descriptionDetails = details[0].trim();
+        String descriptionDeadline = details[1].trim();
```

Comment:
can change the variable name of descriptionDeadline to increase readability, since people may be confused about the difference between descriptionDeadline and deadline description.

Reply:
Thank you for the naming suggestion, will address this issue in my next update!